docs/openapi-docs/src/main/scala/sttp/tapir/docs/openapi/EndpointToOperationResponse.scala

```diff
@@ -22,90 +21,68 @@ private[openapi] class EndpointToOperationResponse(objectSchemas: Schemas, codec
       defaultResponseKey: ResponsesKey,
       defaultResponse: Option[Response]
   ): ListMap[ResponsesKey, ReferenceOr[Response]] = {
-    val responses: ListMap[ResponsesKey, ReferenceOr[Response]] = {
-      val outputsList = output.asBasicOutputsList
-      val byStatusCode = outputsList.groupBy(_._1)
-      ListMap(
-        outputsList
-          .map(_._1)
-          .distinct
-          .flatMap { sc => byStatusCode.get(sc).map((sc, _)) }
-          .flatMap { case (sc, responses) =>
-            // using the "default" response key, if no status code is provided
-            val responseKey = sc.map(c => ResponsesCodeKey(c.code)).getOrElse(defaultResponseKey)
-            responses
-              .flatMap { case (sc, outputs) => outputsToResponse(sc, outputs) }
-              .reduceOption(_.merge(_))
-              .map(response => (responseKey, Right(response)))
-          }: _*
-      )
+    val outputs = output.asBasicOutputsList
+    val byStatusCode = output.asBasicOutputsList.groupBy { case (sc, _) => sc }
+    outputs.map { case (sc, _) => sc }.distinct.flatMap { sc =>
+      val responseKey = sc.map(c => ResponsesCodeKey(c.code)).getOrElse(defaultResponseKey)
+      val allOutputs = byStatusCode.get(sc).map(_.flatMap { case (_, output) => output }).getOrElse(List())
```

Comment:
or maybe, it would make sense to convert `byStatusCode` to a `Map[Option[StatusCode], Vector[EndpointOutput.Basic[_]]]` right in the beginning, then we could juse use `.keys` and `.values`